### PR TITLE
detect and note newline sequence

### PR DESCRIPTION
**Pull request by malaffoon**
This is the hotfix for the problem with newline sequences in student group upload files. There is a matching patch to RDW_Ingest.

The approach is to detect the newline sequence during validation and store it with the file properties. Then the group processor can use that to set the expected newline during import.

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java [hunk 1]
```diff
@@ -82,22 +85,26 @@ public StudentGroupBatch upload(final User user, final MultipartFile file) {
...
-        final List<CsvValidationFailure> validationFailures = new ArrayList<>();
 
         try {
+            final List<CsvValidationFailure> validationFailures = newArrayList();
+            final Map<String, String> fileProperties = newHashMap();
+            fileProperties.put("username", user.getUsername());
+
             try (final DigestInputStream digestStream = new DigestInputStream(file.getInputStream(), messageDigest)) {
-                validationFailures.addAll(validate(digestStream, permissionScope));
+                validate(digestStream, permissionScope, validationFailures, fileProperties);
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java:95]: Modified the signature of the helper to effectively return failures and properties.

File: webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java [hunk 2]
```diff
@@ -82,22 +85,26 @@ public StudentGroupBatch upload(final User user, final MultipartFile file) {
...
                 logger.debug("Validated request: {}", file.getOriginalFilename());
-                archiveFile(file, digest, user);
+                archiveFile(file, digest, fileProperties);
                 studentGroupBatchBuilder
                         .digest(digest)
                         .status(ImportStatus.ACCEPTED);
             } else {
-                logger.debug("Request failed validation: {}", file.getOriginalFilename());
+                logger.info("Request failed validation: {}", file.getOriginalFilename());
+                archiveFile(file, digest, fileProperties);
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java:107]: sneaking in another change: archive the file even if the data is bad

**Review by jtreuting — APPROVED** [review]

File: webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java (hunk 2 above)
Comment (agorina) [reply webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java:107]: Nice!

**Review by agorina — APPROVED** [review]

**Review by wtritch — APPROVED** [review]
LGTM

**Pull request merged**